objects/src/accounts/delta/storage.rs

```diff
@@ -251,6 +274,13 @@ impl StorageMapDelta {
     }
 }
 
+/// Converts a [StorageMap] into a [StorageMapDelta] for initial delta construction.
+impl From<&StorageMap> for StorageMapDelta {
+    fn from(map: &StorageMap) -> Self {
+        Self(map.leaves().flat_map(|(_, leaf)| leaf.entries().into_iter().cloned()).collect())
```

Comment:
Maybe we could do this a bit more manually, but avoid calling `SmtLeaf::entries` which allocates a new `Vec` for every `SmtLeaf::Single`, which would be fairly inefficient for large `StorageMap`s. Something like this would avoid that allocation:

```rust
let mut delta = StorageMapDelta::new(BTreeMap::default());
for (_, leaf) in map.leaves() {
    match leaf {
        SmtLeaf::Empty(_) => continue,
        SmtLeaf::Single((key, value)) => {
            delta.insert(*key, *value);
        },
        SmtLeaf::Multiple(vec) => {
            vec.into_iter().for_each(|(key, value)| delta.insert(*key, *value));
        },
    }
}

delta
```